File: DebugIntro/C/09_Tarpit.c
#include <stdio.h>
#include <float.h>
#include <time.h>
#include <string.h>

int main(int argc, char** argv){

  int i;
  const int reps = 1000;
  FILE *fileptr;
  clock_t start, end;
  int quiet = 0;

  if(argc > 1){
    if(strcmp(argv[1], "-q")) quiet = 1;
  }


 /* Opening once for write truncates any content*/
  fileptr = fopen("./tmp.out", "w");
  fclose(fileptr);
  start = clock();

#ifndef FIX
  for(i = 0; i< reps; i++){
    if(quiet) printf("%d\n", i);
    fileptr = fopen("./tmp.out", "a");
    fprintf(fileptr, "%d ", i);
    fclose(fileptr);
  }
#else
  fileptr = fopen("./tmp.out", "a");
  for(i = 0; i< reps; i++){
    if(quiet) printf("%d\n", i);
    fprintf(fileptr, "%d ", i);
  }
  fclose(fileptr);

#endif

  end = clock();
  printf("Elapsed time %f s\n", ((double) (end - start)) / CLOCKS_PER_SEC);

}

File: DebugIntro/Python/09_Tarpit.py
import time
import sys

if __name__ == "__main__":

  _reps = 1000

  quiet = False
  fix = False

  if '-q' in sys.argv: quiet = True
  if 'fix' in sys.argv : fix = True

  #Open once to remove previous content
  outfile = open('./tmp.out', 'w')
  outfile.close()

  start_time=time.time()

  #------ first path is bad, second a fixed version
  if not fix:
    for i in range(0, _reps):
      if(not quiet): print(i)
      outfile = open('./tmp.out', 'a')
      outfile.write(str(i))
      outfile.close()

  else:
    outfile = open('./tmp.out', 'a')
    for i in range(0, _reps):
      if(not quiet): print(i)
      outfile.write(str(i))

    outfile.close()
  #------------

  elapsed_time=time.time()-start_time

  print("Elapsed time: {0} s".format(elapsed_time))

